presto-hive-common/src/main/java/com/facebook/presto/hive/CacheQuota.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.units.DataSize;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.hash.Hashing.md5;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class CacheQuota
+{
+    private final long identifier;
+    private final Optional<DataSize> quota;
+
+    public static final CacheQuota NO_CACHE_CONSTRAINTS = new CacheQuota("NO_IDENTITY", Optional.empty());
+
+    public CacheQuota(String identity, Optional<DataSize> quota)
+    {
+        this(md5().hashString(identity, UTF_8).asLong(), quota);
```

Comment:
This will be called on the coordinator. One concern is the complexity of `md5()`. I'm not sure if `md5()` is more expensive or shipping the `identity` is..... To be save, I would just move this part of logic to `getIdentifier()` call.

Reply:
Valid point. 

Since HiveSplit already has all the identity information we need, instead of passing Identity(could be expensive) we just pass the Scope info to worker and let worker reuse identity from HiveSplit and calculate the identifier there